src/bin/pg_autoctl/cli_service.c

```diff
@@ -224,6 +195,8 @@ cli_monitor_run(int argc, char **argv)
 			log_warn("Re-establishing connection. We might miss notifications.");
 			pgsql_finish(&(monitor.pgsql));
 
+			ensure_monitor_running(&monitor, &mconfig);
```

Comment:
Maybe the function should return a boolean and we should implement error handling when we fail to start/restart Postgres. Either quit with a non-zero code so that systemd or another supervisor can handle the situation, or implement a restart strategy, or maybe both.